src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -307,26 +307,40 @@ private static string GetRestoreProjectPath(IVsTargetFrameworks tfms)
             return GetNonEvaluatedPropertyOrNull(tfms, RestorePackagesPath, e => e);
         }
 
+        /// <summary>
+        /// The result will contain CLEAR and no sources specified in RestoreSources if the clear keyword is in it.
+        /// If there are additional sources specified, the value RestoreAdditionalProjectSources will be set in the result and then all the additional sources will follow
+        /// </summary>
         private static string[] GetRestoreSources(IVsTargetFrameworks tfms)
         {
             var sources = MSBuildStringUtility.Split(GetNonEvaluatedPropertyOrNull(tfms, RestoreSources, e => e));
 
             sources = HandleClear(sources);
 
             var additional = MSBuildStringUtility.Split(GetNonEvaluatedPropertyOrNull(tfms, RestoreAdditionalProjectSources, e => e));
-            sources = sources.Concat(additional).ToArray();
-
+            if (additional.Length != 0)
+            {
+                sources = (sources.Concat(new string[] { RestoreAdditionalProjectSources }).Concat(additional)).ToArray();
```

Comment:
I dont feel comfortable with this heck and just relying on the order of stuff being added into the array. Could there be a better way maintaining restore sources and restore additional sources? I even won't mind adding separate arrays for these sources, but that should be the last resort.